includes/libs/rdbms/database/DatabaseMysqli.php

```diff
@@ -171,7 +171,10 @@ protected function mysqlConnect( $server, $user, $password, $db ) {
 		} else {
 			$mysqli->options( MYSQLI_SET_CHARSET_NAME, 'binary' );
 		}
-		$mysqli->options( MYSQLI_OPT_CONNECT_TIMEOUT, 3 );
+		$mysqli->options( MYSQLI_OPT_CONNECT_TIMEOUT, $this->connectTimeout ?: 3 );
+		if ( $this->receiveTimeout ) {
+			$mysqli->options( MYSQLI_OPT_READ_TIMEOUT, $this->receiveTimeout );
+		}
```

Comment:
OK - in upstream: https://github.com/Wikia/mediawiki/blob/2e236f056f193aa6702c42ffc4b2e678c2745eeb/includes/libs/rdbms/database/DatabaseMySQL.php#L754-L765